components/Editor.js

```diff
@@ -1179,10 +1179,79 @@ Prism.languages.clike = {
     Prism.languages.markup.tag.addInlined('script', 'javascript'),
   (Prism.languages.js = Prism.languages.javascript);
 
+/* "prismjs/components/prism-reason" */
+
+Prism.languages.reason = Prism.languages.extend('clike', {
+  comment: {
+    pattern: /(^|[^\\])\/\*[\s\S]*?\*\//,
+    lookbehind: true,
+  },
+  string: {
+    pattern: /"(?:\\(?:\r\n|[\s\S])|[^\\\r\n"])*"/,
+    greedy: true,
+  },
+  // 'class-name' must be matched *after* 'constructor' defined below
+  'class-name': /\b[A-Z]\w*/,
+  keyword: /\b(?:and|as|assert|begin|class|constraint|do|done|downto|else|end|exception|external|for|fun|function|functor|if|in|include|inherit|initializer|lazy|let|method|module|mutable|new|nonrec|object|of|open|or|private|rec|sig|struct|switch|then|to|try|type|val|virtual|when|while|with)\b/,
+  operator: /\.{3}|:[:=]|=(?:==?|>)?|<=?|>=?|[|^?'#!~`]|[+\-*\/]\.?|\b(?:mod|land|lor|lxor|lsl|lsr|asr)\b/,
+});
+Prism.languages.insertBefore('reason', 'class-name', {
+  character: {
+    pattern: /'(?:\\x[\da-f]{2}|\\o[0-3][0-7][0-7]|\\\d{3}|\\.|[^'\\\r\n])'/,
+    alias: 'string',
+  },
+  constructor: {
+    // Negative look-ahead prevents from matching things like String.capitalize
+    pattern: /\b[A-Z]\w*\b(?!\s*\.)/,
+    alias: 'variable',
+  },
+  label: {
+    pattern: /\b[a-z]\w*(?=::)/,
+    alias: 'symbol',
+  },
+}); // We can't match functions property, so let's not even try.
+
+delete Prism.languages.reason.function;
+
+/* "prismjs/components/prism-json" */
+
+Prism.languages.json = {
+  property: /"(?:\\.|[^\\"\r\n])*"(?=\s*:)/i,
+  string: {
+    pattern: /"(?:\\.|[^\\"\r\n])*"(?!\s*:)/,
+    greedy: true,
+  },
+  number: /\b0x[\dA-Fa-f]+\b|(?:\b\d+\.?\d*|\B\.\d+)(?:[Ee][+-]?\d+)?/,
+  punctuation: /[{}[\]);,]/,
+  operator: /:/g,
+  boolean: /\b(?:true|false)\b/i,
+  null: /\bnull\b/i,
+};
+Prism.languages.jsonp = Prism.languages.json;
+
+const extensions = new Map([
+  ['js', 'javascript'],
+  ['re', 'reason'],
+  ['json', 'json'],
+]);
+
+const extensionsReturner = extension => {
+  try {
+    return extensions.get(extension);
+  } catch {
+    return 'javascript';
+  }
+};
+
 export default props =>
   html`
+    ${console.log(props.extension)}
     <${Editor}
-      highlight=${code => Prism.highlight(code, Prism.languages.js)}
+      highlight=${code =>
+        Prism.highlight(
+          code,
+          Prism.languages[extensionsReturner(props.extension)]
```

Comment:
Here I would do `extensions[props.extension] || 'javascript'` instead of try catching; where extensions is just an object, rather than a Map. Unless I am missing something?

Reply:
Yep done!